crates/walrus-service/src/node/storage/database_config.rs

```diff
@@ -558,10 +507,253 @@ impl Default for DatabaseConfig {
             committee_store: None,
             event_store: None,
             init_state: None,
+            shared_shard_storage_block_cache_config: None,
         }
     }
 }
 
+/// A factory for creating RocksDB Options from DatabaseTableOptions with a shared block cache.
+/// This ensures all column families share the same block cache instance.
+#[derive(Clone)]
+pub struct DatabaseTableOptionsFactory {
+    /// The shared block cache instance for all column families.
+    shared_shard_storage_block_cache: Option<Cache>,
+    /// The database configuration.
+    config: DatabaseConfig,
+}
+
+impl DatabaseTableOptionsFactory {
+    /// Creates a new DatabaseTableOptionsFactory with a shared block cache.
+    /// If `is_storage_db` is true, the database table config is for storage DB, and therefore
+    /// the shared block cache will be used for shard storage column families if it is set.
+    pub fn new(config: DatabaseConfig, is_storage_db: bool) -> Self {
+        let shared_shard_storage_block_cache = if is_storage_db {
```

Comment:
Do we really need `is_storage_db` flag here? Isn't the config `shared_shard_storage_block_cache_config` enough to control whether we use shared cache or not.

Reply:
yeah, I added more detailed comments to explain this. The idea is that even if shared_shard_storage_block_cache_config is set, we don't want to apply shared block cache in event and event blob writer DB. The reasoning is in the reply above (from Aug 22)